library/src/androidTest/java/org/xmtp/android/library/GroupTest.kt

```diff
@@ -1022,4 +1023,54 @@ class GroupTest {
         // Next syncAllGroups will not include the inactive group
         assertEquals(numGroups, 2u)
     }
+
+    @Test
+    fun testGroupDisappearingMessages() = runBlocking {
+        // Can enable message expiration on create
+        val setting = MessageDisappearingSettings(0L, 1L)
+        val boGroup = boClient.conversations.newGroup(
+            listOf(alix.walletAddress),
+            messageDisappearingSettings = setting
+        )
+
+        val messageId = boGroup.send("howdy")
+        alixClient.conversations.syncAllConversations()
+
+        val message = boClient.findMessage(messageId)
+        val alixGroup = alixClient.findGroup(boGroup.id)
+
+        assertEquals(boGroup.messageDisappearingSettings()?.disappearDurationInNs, setting.disappearDurationInNs)
+
+        assertEquals(boGroup.messages().size, 2)
+        assertEquals(alixGroup?.messages()?.size, 1)
+
+        boGroup.updateMessageDisappearingSettings(MessageDisappearingSettings(message!!.sentAtNs, 1L))
+        boGroup.send("howdy2")
+        alixGroup!!.send("howdy howdy")
+
+        assertEquals(boGroup.messages().size, 4)
+        assertEquals(alixGroup.messages().size, 4)
+        boClient.conversations.syncAllConversations()
+        alixClient.conversations.syncAllConversations()
+        Thread.sleep(4000)
+
+        assertEquals(boGroup.messages().size, 2)
```

Comment:
@mchenani I tried to match this as closely to the rust test but it's still not working like expected?